pkg/rdt/rdt.go

```diff
@@ -87,6 +90,30 @@ type MonGroup interface {
 	Parent() CtrlGroup
 }
 
+// Monitor is the interface for retrieving RDT monitoring data
+type Monitor interface {
+	GetMonData() MonData
+}
+
+// MonData contains monitoring stats of one monitoring group
+type MonData struct {
+	L3 MonL3Data
+}
+
+// MonL3Data contains L3 monitoring stats of one monitoring group
+type MonL3Data map[uint64]MonLeafData
+
+// MonLeafData represents the raw numerical stats from one RDT monitor data leaf
+type MonLeafData map[string]uint64
+
+// MonResource is the type of RDT monitoring resource
+type MonResource string
```

Comment:
nitpick: this is a general note, much more related to my own code than this...

I've noticed lately that I tend to often do the same thing you do here. Define a set of `const`s with `string`s as the underlying type, while the consts are also used to index `map`s. With this setup, the only thing you gain/save is typing: you don't have to write a trivial `String()` function with a `receiver` of your defined type. The price you pay for that laziness is much less efficient `map lookup`s. Sometimes, maybe even often, this does not matter at all. However...

In recently written code of my own, I prefer to use `int` as the underlying type of the `const`s with a `String()` defined for it with the right effect. IOW, I prefer to spend extra time (typically) in `formatted printing`/string operations, which are supposed to be rare and heavy anyway, then in `map lookup`s which tend to be more frequent and therefore should be more efficient.

Reply:
Good point. And, yes, as an old C developer I've noticed this pattern and it's implications, too 😄 I've adapted this habit from K8s et al I guess. But, I think it's handy in some cases by reducing the LOC.

In this particular case the string type is used in a very non-performance-critical code path (and, I don't see it ending up in such a code path). Thus, I let this be as it is, for now.